libmapi/idset.c

```diff
@@ -360,11 +360,17 @@ static void check_idset(const struct idset *idset)
 _PUBLIC_ struct idset *IDSET_parse(TALLOC_CTX *mem_ctx, DATA_BLOB buffer, bool idbased)
 {
 	struct idset		*idset, *prev_idset = NULL;
-        DATA_BLOB		guid_blob, globset;
-	uint32_t		total_bytes, byte_count;
+	DATA_BLOB		guid_blob, globset;
+	uint32_t		total_bytes, byte_count, id_length;
 
 	if (buffer.length < 17) return NULL;
 
+	if (idbased) {
+		id_length = 2;
+	} else {
+		id_length = 16;
+	}
+
```

Comment:
Can you turn this block of code into a ternary statement instead?

id_length = (idbased == true) ? 2 : 16;